lib/perl/Genome/ModelDeprecated.pm

```diff
@@ -631,38 +631,53 @@ sub _resolve_type_name_for_class {
 
 sub compatible_instrument_data {
     my $self = shift;
-    my %params;
 
-    if (my @samples = $self->get_all_possible_samples)  {
-        my @sample_ids = map($_->id, @samples);
-        %params = (
-                   sample_id => \@sample_ids,
-               );
+    if($self->analysis_project) {
+        my $config_item = $self->analysis_project_bridge->config_profile_item;
+        unless($config_item) {
+            die $self->error_message('Configuration missing from analysis project association.  Cannot determine compatible instrument data for model %s and analysis project %s.', $self->__display_name__, $self->analysis_project->__display_name__);
+        }
+
+        my $rmm = Genome::Config::Translator->get_rule_model_map_from_config($config_item);
+        my @project_instrument_data = $self->analysis_project->instrument_data;
```

Comment:
It's an interesting trade-off.  The most likely scenario is that we'll be processing many instrument-data sequentially from the same project, and so we'd end up loading them repeatedly.  But since it's likely we will have several large projects at once I think you're right that we shouldn't blithely load them all like this.

Would it be worthwhile to try to turn the rules into a BoolExpr and limit the amount of instrument data we're loading in the first place?